### PR TITLE
[WIP] Add doc search

### PR DESCRIPTION
**Pull request by helfi92**
Fixes https://github.com/taskcluster/taskcluster/issues/450.

### CODE REVIEW

**Comment by helfi92** [comment]
In order to run the crawler, I:

1. Created a file `doc-env` with content:

```bash
APPLICATION_ID=FUFMXT38CF
API_KEY=8ddc2f77936ff02ce46e8db68c6b14ee
```

2. Ran the docker image

```bash
brew install jq
docker run -it --env-file=doc-env -e "CONFIG=$(cat docsearch.json | jq -r tostring)" algolia/docsearch-scraper
```

File: ui/src/components/DocSearch/index.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+export default class DocSearch extends Component {
+  static propTypes = {
+    onSubmit: func.isRequired,
+  };
+
+  componentDidMount() {
+    docsearch({
+      apiKey: 'c7d22dbe731b27772278220687a87208',
+      // Used for identification when using Algolia's API
```
Comment (djmitche) [inline ui/src/components/DocSearch/index.jsx]: Is it OK to make this information public?  If this is all generated locally, how is the API involved?  Could someone else use this key and ID to break our search?  Or possibly to inject malicious code into our users' browser sessions?

**Comment by helfi92** [comment]
> Does that end up spidering the entire ui (for example if a doc links to /hooks, does it then spider every hook?)

[stop_urls](https://community.algolia.com/docsearch/config-file.html#stop_urls-optional) can be set to only scan and index links that fall under `/docs`.

> When we add a schema viewer that recursively expands $ref, will it expand those infinitely?

Any links not pertaining to the same domain will automatically not be scanned. `stop_urls` can also be used which accepts a regex.

>  Where does the index end up?

In Algolia servers

>  If it's on the algolia servers, then we'll need to have a disctinct index per Taskcluster build, and somehow reference that from the build (maybe a different API key per build?).

I belive the [`index_name`](https://community.algolia.com/docsearch/config-file.html#index_name) should be able to handle that.

> 10 minutes seems really long

Agreed. Note that the indexing can happen asynchronously in the background and doesn't necessarily need to block the `yarn build` command. Perhaps it can be run as a task?

**Comment by djmitche** [comment]
From talking amongst the team just now, this takes too long to index and we'll look to take a simpler, less-fancy but faster approach.

**Pull request closed without merging**